### PR TITLE
Add Batch Norm

### PR DESCRIPTION
**Pull request by Ziminli**
- Add the CPU and CUDA implementation and the frontend python test for Batch Normalization 
 - modified f32_to_f16 in common_cpu for more accurate behavior

### CODE REVIEW

File: src/ops/batch_norm/cuda/batch_norm.cc [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    uint64_t ndim = y->ndim;
+    if (ndim != x->ndim || scale->ndim != b->ndim || scale->ndim != mean->ndim || scale->ndim != var->ndim) {
+        return STATUS_BAD_TENSOR_SHAPE;
+    }
+    for (size_t i = 0; i < ndim; ++i) {
+        if (y->shape[i] != x->shape[i]) {
+            return STATUS_BAD_TENSOR_SHAPE;
+        }
+    }
+    for (size_t i = 0; i < scale->ndim; ++i) {
```
Comment (kilinchange) [inline src/ops/batch_norm/cuda/batch_norm.cc:23]: 直接检查 scale->ndim==1 且 scale->shape[0] == x->shape[1] 就行，不用循环

File: src/ops/batch_norm/cpu/batch_norm_cpu.cc [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    uint64_t ndim = y->ndim;
+    if (ndim != x->ndim || scale->ndim != b->ndim || scale->ndim != mean->ndim || scale->ndim != var->ndim) {
+        return STATUS_BAD_TENSOR_SHAPE;
+    }
+    for (size_t i = 0; i < ndim; ++i) {
+        if (y->shape[i] != x->shape[i]) {
+            return STATUS_BAD_TENSOR_SHAPE;
+        }
+    }
+    for (size_t i = 0; i < scale->ndim; ++i) {
```
Comment (kilinchange) [inline src/ops/batch_norm/cpu/batch_norm_cpu.cc:23]: 直接检查 scale->ndim==1 且 scale->shape[0] == x->shape[1] 就行，不用循环

**Review by kilinchange — CHANGES_REQUESTED** [review]

**Comment by kilinchange** [comment]
在 operators/include/infini_operators.h 加一下头文件引用

**Review by kilinchange — APPROVED** [review]

File: operatorspy/tests/batch_norm.py [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+
+
+def test_bang(lib, test_cases):
+    import torch_mlu
+
+    device = DeviceEnum.DEVICE_BANG
+    handle = create_handle(lib, device)
+    for x_shape, eps, inplace in test_cases:
+        test(lib, handle, "mlu", x_shape, eps, tensor_dtype=torch.float16, inplace=inplace)
+        test(lib, handle, "mlu", x_shape, eps, tensor_dtype=torch.float32, inplace=inplace)
```
Comment (PanZezhong1725) [inline operatorspy/tests/batch_norm.py:179]: 把type也作为test case中的一个参数吧

**Review by PanZezhong1725 — CHANGES_REQUESTED** [review]

**Review by PanZezhong1725 — CHANGES_REQUESTED** [review]
